deepchem/models/torch_models/dtnn.py

```diff
@@ -0,0 +1,234 @@
+import numpy as np
+from typing import List
+
+import torch.nn as nn
+import torch
+
+from deepchem.models.losses import L2Loss
+from deepchem.models.torch_models import layers
+from deepchem.models.torch_models import TorchModel
+from deepchem.data.datasets import Dataset
+
+
+class DTNN(nn.Module):
+    """Deep Tensor Neural Networks
+
+    In this class, we establish a sequential model for the Deep Tensor Neural Network (DTNN) [1]_.
+
+    Examples
+    --------
+    >>> import os
+    >>> import torch
+    >>> from deepchem.models.torch_models import dtnn
+    >>> from deepchem.data import SDFLoader
+    >>> from deepchem.feat import CoulombMatrix
+    >>> from deepchem.models.torch_models import dtnn
+    >>> # Get Data
+    >>> model_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    >>> dataset_file = os.path.join(model_dir, 'tests/assets/qm9_mini.sdf')
+    >>> TASKS = ["alpha", "homo"]
+    >>> loader = SDFLoader(tasks=TASKS, featurizer=CoulombMatrix(29), sanitize=True)
+    >>> data = loader.create_dataset(dataset_file, shard_size=100)
+    >>> inputs = dtnn._compute_features_on_batch(data.X)
```

Comment:
you should not define `_computer_features_on_batch` to be a private helper function. It should be something like DTNN_Featurizer, like this
`featurizer = DTNN_Featurizer(num_tasks=data.y.shape[0]`
`inputs = featurizer(data.X)`
In this way this featurization can be used by other models in a standardized way.